src/shims/x86/sse.rs

```diff
@@ -518,27 +521,36 @@ fn unary_op_f32<'tcx>(
         }
         FloatUnaryOp::Rcp => {
             let op = op.to_scalar();
-            // Use 1.0001 as a crude way of simulating inaccuracy of rcp
-            // 1.0001 / x ~= 1 / x
-            let one = Single::from_bits(1.0001f32.to_bits().into());
-            // floating point operation without side effects
-            #[allow(clippy::arithmetic_side_effects)]
+            // Calculate an approximation of 1 / x as (1 + err) / x, where
+            // err is a random value with a magnitude in the order of 2^-12
+            let one = (Single::from_u128(1).value + gen_random_float_error(this, -12)).value;
```

Comment:
Shouldn't it apply the error *after* doing the computation?

Reply:
Applying the error after doing the computation would result in an expression like `(1 / x) * (1 + err)`, which I simplified to `(1 + err) / x`.

Reply:
I'd prefer to do the obviously correct thing consistently (here and for sqrt) over being clever -- unless there is some good reason for having to be clever.

Reply:
Done, there is not really a reason :)